Modules/Devices/src/Devices.Domain/Entities/Identities/IdentityDeletionProcess.cs

```diff
@@ -126,4 +136,13 @@ public void Approve(IdentityAddress address, DeviceId approvedByDevice)
         Approve(approvedByDevice);
         _auditLog.Add(IdentityDeletionProcessAuditLogEntry.ProcessApproved(Id, address, approvedByDevice));
     }
+
+    public void Cancel(IdentityAddress address, DeviceId cancelledByDevice)
+    {
+        if (Status != DeletionProcessStatus.Approved)
+            throw new DomainException(DomainErrors.NoDeletionProcessWithRequiredStatusExists());
+
+        Cancel(cancelledByDevice);
```

Comment:
Out of curiosity: why did you extract this into a separate method?

Reply:
To be honest I took a look at the `Approve` method and followed the same pattern.

Reply:
The `Approve´ method is a special case, because it is called multiple times. `Cancel` instead is only called once.

Reply:
I understand. I have removed the private `Cancel` method and put the logic inside the public `Cancel` method.